### ISSUE TITLE
Domain picker: searching for FQDNs does not always return the TLD specified

### ISSUE THREAD

**Issue by roo2** [Automattic/wp-calypso#41066]
In the domain picker, when you're searching for a certain domain name including the TLD such as 
`blogadyblogblagadoo.blog`, the search results should include the domain name that we searched for. In the attached screenshot, you aren't given the option to select the .blog TLD.

<img width="446" alt="Screen Shot 2020-04-14 at 7 49 04 AM" src="https://user-images.githubusercontent.com/22446385/79178960-ecdf7000-7e49-11ea-8fce-58c18051ffa3.png">

#### Steps to reproduce
On https://wordpress.com/new fill out the details until the domain picker is enabled and open it.
Search for a domain including the tld part e.g. `blogadyblogblagadoo.blog`

**Comment by lancewillett** [comment]
I believe this looks like a duplicate of #41063.

**Comment by klimeryk** [comment]
No, this appears to be a regression from current signup - if a user searches for a fully qualified domain name (FQDN), for example, `example.com`, we should _always_ check for the availability of this domain - that's the expectation of the user. And we need to guide the user depending on its availability, if it's mappable, transferable, etc. Happy to go into more details, but it's too much for a GitHub issue :)

Example of such a case: `rata.mx` is a domain I own outside WordPress.com, I've searched for it in signup and I get a message that it's taken, but it's transferable - and the option to do so:

<img width="1002" alt="Screenshot 2020-04-15 at 11 32 52" src="https://user-images.githubusercontent.com/3392497/79332797-f8e03600-7f0c-11ea-969b-ff0c0c4a4fb2.png">

**Comment by razvanpapadopol** [comment]
@alshakero did some progress in this direction in #48550

**Comment by razvanpapadopol** [comment]
@alshakero @ciampo can we consider this issue fixed everywhere (including `/new`) or https://github.com/Automattic/wp-calypso/pull/48550 addressed only domain from cart -> selected domain in Focused Launch?

**Comment by alshakero** [comment]
Sadly it was not fixed in the domain picker (which is everywhere except `/start`). As you said, what was fixed is _only domain from cart -> selected domain in Focused Launch_. 

It might be worth mentioning that #48550 had the potential to be related to this before, but it's no longer related. Before, we used to query the domain suggestions API with the domain name from the cart, and pick the matching domain and list it as one of the suggestions. This made this issue a blocker for us as we depended on the domain suggestions API to return the exact match.

But in the end we moved to the availability endpoint instead, to eliminate the searching part and query for the exact domain. The availability endpoint gave us all the info we needed to convert the cart domain into a virtual domain suggestion. It also gave us the availability info which made even more sense.
